### PR TITLE
Prepare virtual channels

### LINKED ISSUE #9: Support for virtual channels
A virtual channel is a channel that has potentially different participants than its parent channels.

Currently we assume that a sub-channel of a parent channel always has the same participants.

In order to prepare for support of virtual channels, we need to allow sub-channels with different participants.

More details in hyperledger-labs/go-perun#82.

### PR DESCRIPTION
**Pull request by matthiasgeihs**
Closes #9

### CODE REVIEW

**Review by sebastianst — CHANGES_REQUESTED** [review]
An intro meeting would be good as there are quite some substantial changes. But generally looks very good.

We also need to hash in the `bool ledgerChannel` for the `channelId` calculation, or users can just set it as they want, no?

File: contracts/Channel.sol [hunk 1]
```diff
@@ -27,6 +27,7 @@ library Channel {
         address[] participants;
         address app;
         bool ledgerChannel;
+        bool virtualChannel;
```
[the remark this answers is not in the document]
Comment (matthiasgeihs) [reply contracts/Channel.sol]: I agree, it's not super clean right now and using Enums might be better if we keep that distinction.

However, it should be possible to go back to only a single flag `ledgerChannel` once we implement the protocols for virtual channels with sub-channels in go-perun, so I prefer to keep as is for now.

**Comment by matthiasgeihs** [comment]
> Please open an issue regarding virtual sub-app-channels, which we should add at some point.

Done, #12

File: contracts/Channel.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebastianst) [reply contracts/Channel.sol]: OK let's do it in #12 if we realize we need it.

**Review by sebastianst — APPROVED** [review]
LGTM

**Pull request merged**